loki/transformations/loop_blocking.py

```diff
@@ -0,0 +1,121 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+from loki import Subroutine, ir, FindNodes, Assignment, Sum, Product, Loop, LoopRange, IntLiteral, \
+    Quotient, InlineCall, DeferredTypeSymbol, Transformer, Array, Variable, RangeIndex, Scalar, \
+    SubstituteExpressions, FindVariables
+
+
+def split_loop(routine: Subroutine, loop: ir.Loop, block_size: int):
+    """
+    Assumes that loop is contiguous in blocking variable
+    into a blocked loop in those variables. Does not block
+    the loop data
+    """
+    # blocking variable declarations
+    decls = FindNodes(ir.VariableDeclaration).visit(routine.spec)
+    block_size_var = loop.variable.clone(name=loop.variable.name + '_loop_block_size')
+    num_blocks_var = loop.variable.clone(name=loop.variable.name + '_loop_num_blocks')
+    outer_loop_var = loop.variable.clone(name=loop.variable.name + '_loop_outer_idx')
+    inner_loop_var = loop.variable.clone(name=loop.variable.name + '_loop_local')
+    global_loop_var = loop.variable
+    block_start = loop.variable.clone(name=loop.variable.name + '_loop_block_start')
+    block_end = loop.variable.clone(name=loop.variable.name + '_loop_block_end')
+
+    routine.variables += (block_size_var, num_blocks_var, outer_loop_var, block_start, block_end)
+
+    # Inner loop
+    inner_body = Assignment(global_loop_var, Sum(children=(
+        Product(children=(outer_loop_var, block_size_var)), inner_loop_var)))
```

Comment:
This is perfectly fine and easy enough in this case, just to point out that you can also create expressions by parsing strings on-the-fly using [`parse_expr`](https://sites.ecmwf.int/docs/loki/main/loki.expression.parser.html#module-loki.expression.parser). This is a relatively recent addition, hence you don't find many examples in the existing code, yet, but can avoid the tedious construction of expressions.
Here, this could look like this:

```suggestion
    inner_body = Assignment(
        lhs=global_loop_var,
        rhs=parse_expr(f'{outer_loop_var} * {block_size_var} + {inner_loop_var}', scope=routine)
    )
```